### PR TITLE
Change graveler iterator interface

### PR DESCRIPTION
**Pull request by itaiad200**

### CODE REVIEW

File: graveler/pgiterator.go [hunk 1]
```diff
@@ -109,10 +123,12 @@ func (ri *PGRepositoryIterator) SeekGE(id RepositoryID) bool {
 	} else {
 		ri.buf = make([]*RepositoryRecord, 0)
 	}
-	return true
 }
 
 func (ri *PGRepositoryIterator) Value() *RepositoryRecord {
+	if ri.postInit || ri.postSeekGE {
```
Comment (guy-har) [inline graveler/pgiterator.go:129]: It would be nice to also return nil in case of error

File: graveler/pgiterator.go [hunk 2]
```diff
@@ -91,14 +104,15 @@ func (ri *PGRepositoryIterator) fetch(initial bool) {
...
 
-func (ri *PGRepositoryIterator) SeekGE(id RepositoryID) bool {
+func (ri *PGRepositoryIterator) SeekGE(id RepositoryID) {
 	ri.offset = string(id)
 	ri.shouldFetch = true
 	ri.buf = make([]*RepositoryRecord, 0)
 	ri.fetch(true) // do a new initial fetch
-
+	ri.postSeekGE = true
+	ri.postInit = false
```
Comment (guy-har) [inline graveler/pgiterator.go:113]: If SeekGE doesn't return any value, doesn't make more sense to move the fetch into next?

**Review by guy-har — APPROVED** [review]
Looks good, Thanks

File: graveler/pgiterator.go (hunk 2 above)
Comment (itaiad200) [reply graveler/pgiterator.go]: It does, fixed.

File: graveler/pgiterator.go (hunk 1 above)
Comment (itaiad200) [reply graveler/pgiterator.go]: Done.

File: graveler/pgiterator.go [hunk 3]
```diff
@@ -251,6 +243,11 @@ type PGCommitIterator struct {
 	ctx          context.Context
 	repositoryID RepositoryID
 
+	// initPhase turns true when the iterator was created or `SeekGE()` was called.
+	// initPhase turns false when Next() is called.
+	// When initPhase is true, Value() should return nil.
+	initPhase bool
```
Comment (ozkatz) [inline graveler/pgiterator.go:249]: why not use the value? it's nil when you initialize, it makes sense to set it to nil when you `SeekGE`

**Review by ozkatz — APPROVED** [review]

**Pull request merged**